python/lsst/meas/extensions/gaap/gaap.py

```diff
@@ -0,0 +1,186 @@
+        modelPsf : `lsst.afw.detection.GaussianPsf`
+            Target PSF to which to match.
+        footprint : `lsst.afw.detection.Footprint`
+            Footprint of the source to measure.
+
+        Returns
+        -------
+        convExp : `lsst.afw.image.Exposure`
+            Sub-image containing the source, convolving to the target seeing
+        """
+
+        bbox = footprint.getBBox()
+        pixToGrow = self.config.modelPsfMatch.kernel.active.detectionConfig.fpGrowPix
+        # This is conservative for now.
+        # An alternate option is 2*max(self.psfMatch.kConfig.sizeCellX, self.psfMatch.kConfig.sizeCellY)
+        bbox.grow(pixToGrow)  # gets too big
+
+        origPsf = exposure.getPsf()
+
+        maskedImage = exposure.getMaskedImage()
+        subImage = maskedImage.Factory(maskedImage, bbox)
+        subExposure = afwImage.ExposureF(subImage)
+        subExposure.setPsf(origPsf)
+
+        config = self.config.modelPsfMatch
+        config.setDefaults()
+        config.kernel.active.alardSigGauss = [modelPsf.getSigma()]  # The size has to be set dynamically
+        task = config.target(config=config)
+        result = task.run(exposure=subExposure, referencePsfModel=modelPsf)
+        # TODO: DM-27407 will re-Gaussianize the exposure to make the PSF even more Gaussian-like
+        convolved = result.psfMatchedExposure
+        # Setting nans to zeros in a hacky way! DM-28740 will fix this properly.
+        convolved.image.array[np.isnan(convolved.image.array)] = 0.0
+        return convolved
+
+    def measure(self, measRecord: lsst.afw.table.SourceRecord, exposure: afwImage.Exposure,
+                center: lsst.geom.Point2D) -> None:
+        """Measure the GAaP flux
+        """
+
+        psf = exposure.getPsf()
+        if psf is None:
+            raise lsst.meas.base.MeasurementError("No PSF in exposure")
+
+        seeing = psf.computeShape(center).getTraceRadius()
+        for sF in self.config.scalingFactors:
+            if sF < 1.0:
+                # Do not allow the PSF matching to go into deconvolution mode.
+                raise lsst.meas.base.FatalAlgorithmError("The scaling factor has to be greater than"
+                                                         "or equal to 1")
+
+            target = sF*seeing
+            modelPsf = afwDetection.GaussianPsf(65, 65, target)
+            try:
+                convolved = self.convolve(exposure, modelPsf, measRecord.getFootprint())
+            except RuntimeError:
+                continue
+
+            for sigma in self.config.sigmas:
+                effShape = afwGeom.Quadrupole(sigma**2, sigma**2, 0.0)
+                # Assume effShape is elliptical, although it is explicitly circular
+                if target**2 >= min(effShape.getIxx(), effShape.getIyy()):
+                    raise lsst.meas.base.MeasurementError("The GaussianPsf was larger than the effective"
+                                                          "aperture")
```

Comment:
This is another case where we should set a flag field and `continue` instead of raising.

Normally, `MeasurementError` is actually designed to help with that - if you raise it with information about what flag to set, the measurement framework and/or the base class `fail` (I forget which) will set that flag and not warn, instead of warning as it would with most other exceptions.  But:
 - you haven't passed any information about which flag to set when constructing the `MeasurementError` - which should itself be an error, so either this check hasn't been tested or the pybind11 wrappers for it are broken somehow;
 - we don't want to use it here, because we really want to go on to the next iteration of a `for` loop, rather than abort the plugin entirely.  And it's also perfectly (and arguably easier to follow) to set a flag and `continue` or `return` explicitly.